### PR TITLE
Feature/more levels

### PR DESCRIPTION
**Pull request by JoonasHeinonen**
## Changes:
 - Added mission parameters.
 - More tiles.
 - Sprite sheets enhanced.
 - Added sprite sheets for Nefarious, F44, mission parameter card.
 - Optimized code.

### CODE REVIEW

File: src/scripts/npc/npc.gd [hunk 1]
```diff
@@ -25,53 +25,46 @@ func _ready():
...
+						Globle.arena_menu_open = false
+						commence_dialog('timeline_Angela_shark_man')
+						shark_man_dialog_value = process_dialog_value(npc_angela_rivet_dialog_value, 2)
+					(1):
+						shark_man_dialog_value = process_dialog_value(npc_angela_rivet_dialog_value, 0)
+						Globle.player_active = true
+						Globle.arena_menu_open = true
+			"F44":
+				print("Ce soir la vie n'est plus un enfer pour 1 fois.")
+	if (Globle.player_character == "Rivet" and get_node_or_null('DialogNode') == null and Input.is_action_just_pressed("ui_accept") and active):
```
Comment (NathanCenter) [inline src/scripts/npc/npc.gd:64]: when trying to go to angela as rivet it is not activating for some reason.
image

File: src/scripts/world/levels/level_data.gd [hunk 2]
```diff
@@ -13,6 +15,9 @@ func _ready():
 	default_music_stream.loop = true
 	default_music.set_stream(default_music_stream)
 
+	## TODO Use the following pattern to add new mission parameters.
+	mission_params.merge({ "newMission": false })
+
 	add_child(default_music)
 	player = get_node("player")
 	set_music(default_music, default_music_stream)
```
Comment (NathanCenter) [inline src/scripts/world/levels/level_data.gd:25]: There is no music... did we add music to the level?

**Comment by NathanCenter** [comment]
https://github.com/user-attachments/assets/017ea03d-29bf-4051-9a8e-a8d0c9eae156
Odd glitch where one side of sliding is not working for the staircase.

**Comment by NathanCenter** [comment]
image
Why is the sprite so dark in the area level?

**Comment by NathanCenter** [comment]
image
Are you going to move this to the boss and when he dies it drops this key card?

**Review by NathanCenter — CHANGES_REQUESTED** [review]

File: src/scripts/npc/npc.gd [hunk 3]
```diff
@@ -59,6 +59,8 @@ func _input(_event):
 								shark_man_dialog_value = process_dialog_value(npc_angela_rivet_dialog_value, 0)
 								Globle.player_active = true
 								Globle.arena_menu_open = true
+					"F44":
```
[the remark this answers is not in the document]
Comment (JoonasHeinonen) [reply src/scripts/npc/npc.gd]: Later.

File: src/scripts/npc/npc.gd [hunk 4]
```diff
@@ -5,21 +5,20 @@ class_name NPC
...
 var min : int = 0
 var max : int = 8
+
 var player
 var dialog
 
 func _ready():
-	# why so many get parents
 	if (character_name == "Girdeux"):
 		player = get_parent().get_parent().get_parent().get_parent().get_node('player')
```
[the remark this answers is not in the document]
Comment (JoonasHeinonen) [reply src/scripts/npc/npc.gd:22]: You're right. I discovered a better way to find the player node: `player = get_tree().get_root().get_node("Level/player")`

File: scenes/Levels/test_level.tscn [hunk 5]
```diff
@@ -53,7 +52,9 @@ size = Vector3(2.2914, 2, 2)
 script = ExtResource("1_ksfwo")
 z_axis = -1.5
 default_music_stream = ExtResource("2_udr33")
-boss_music_stream = ExtResource("3_xfu21")
+mission_params = {
+"retrieveKeyCard": false
```
[the remark this answers is not in the document]
Comment (JoonasHeinonen) [reply scenes/Levels/test_level.tscn]: Yes, it was in the menu, but not visible, yet.

**Comment by NathanCenter** [comment]
Everything looks good to me!

**Review by NathanCenter — APPROVED** [review]

**Pull request merged**